public/user_data.html

```diff
@@ -1,261 +1,320 @@
+  document.getElementById('androidVersionSPLButton')
+   .addEventListener("click", function() { sendRequest('androidVersionSPL', 'spls?', false);});
+  document.getElementById('bulletinidBULLETINButton')
+   .addEventListener("click", function() { sendRequest('bulletinidBULLETIN', 'bulletins?', false);});
+  document.getElementById('androidVersionBULLETINButton')
+   .addEventListener("click", function() { sendRequest('androidVersionBULLETIN', 'bulletins?', false);});
+  document.getElementById('supportedAndroidVersionsButton')
+   .addEventListener("click", function() { sendRequest('supportedAndroidVersions', 'supportedAndroidVersions', false);});
+  document.getElementById('v1v2bulletinButton')
+   .addEventListener("click", function() { sendRequest('v1v2bulletin', 'cves?', true);});
+ 
+  document.addEventListener('change', displayFileName);
+ 
+  document.getElementById('gcloudToken').addEventListener('click', scrollOnClick);
+ 
+  function scrollOnClick() {
+    window.scrollTo(0, document.body.scrollHeight);
+  }
+ 
+function displayFileName() {
+  const fileNameContainer = document.getElementById('fileName');
+  try{
+   const fileName = document.getElementById('userFile').files[0];
+   let fileNameContainer = document.getElementById('fileName');
+   fileNameContainer.innerHTML = fileName.name;
+  }catch{ fileNameContainer.innerHTML = 'Upload a file';}
+} 
+ 
+function giveAdmin() {
+  generateToken();
+}
+ 
+function generateToken() {
     firebase.auth().onAuthStateChanged(function(user) {
-        const userToken = document.getElementById('tokenContainer').value;
-        const gcloudToken = document.getElementById('gcloudToken').value;
-        const options = {
-            method: 'GET',
-            headers: {
-                'Authorization': 'Bearer ' + gcloudToken,
-                'userToken' : userToken,
-                'Content-Type': 'application/json'
-            }
-        };
-        fetch('/grantAdminRole', options);
+      if (user){
+        firebase.auth().currentUser.getIdToken(true).then(function(idToken) {
+          document.getElementById("tokenContainer").innerHTML = idToken;
+        });
+      }
     });
   }
-  
+ 
+// DO NOT DEPLOY THIS FUNCTION TO PRODUCTION!
+function sendUserToken() {
+   firebase.auth().onAuthStateChanged(function(user) {
+       const userToken = document.getElementById('tokenContainer').value;
+       const gcloudToken = document.getElementById('gcloudToken').value;
+       const options = {
+           method: 'GET',
+           headers: {
+               'Authorization': 'Bearer ' + gcloudToken,
+               'userToken' : userToken,
+               'Content-Type': 'application/json'
+           }
+       };
+      fetch('/grantAdminRole', options).then((response) => response.json()).then((function (output) {
+        if (output["value"] === "User has admin privileges") {
+          document.getElementById('uploadBlock').style.display = "block";
```

Comment:
we should avoid adding the button to the DOM entirely if the user does not have admin privilleges. Someone could easilly edit the element styles/html attributes via developer console to use the button